llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -192,6 +249,20 @@ static void deleteGlobalObject(GlobalObject *GO) {
   }
 }
 
+static void pruning(GlobalObject &GO) {
```

Comment:
What does this function do? There is `doPruning()` and `pruning()`: similarly named but one calls the other.

Reply:
> What does this function do? There is doPruning() and pruning(): similarly named but one calls the other.

`doPruning(Module &M)` is a top level function for the Module. The function covers:
1) It checks all GOs whether the GOs can be pruned or not.
2) If a GO can be pruned and removed, it will be deleted.
3) If a GO can be pruned but not removed, it will be pruned.

`pruning(GlobalObject &GO)` is a function for a GO. When the GO can be pruned and/or removed (the above case 2) and case 3)), the function (`pruning`) is called by the function `doPruning` to prune or remove the GO.

I have changed function name from `pruning` to `removeOrPrune`.